pkg/deployments/tasks/20230427-gauge-working-balance-helper/test/L1GaugeHelper.fork.ts

```diff
@@ -0,0 +1,171 @@
+    task = new Task('20230427-gauge-working-balance-helper', TaskMode.TEST, getForkedNetwork(hre));
+    await task.run({ force: true });
+    workingBalanceHelper = await task.deployedInstance('GaugeWorkingBalanceHelper');
+  });
+
+  describe('getters', () => {
+    it('stores the veDelegationProxy', async () => {
+      expect(await workingBalanceHelper.getVotingEscrowDelegationProxy()).to.equal(veDelegationProxy.address);
+    });
+
+    it('stores the votingEscrow', async () => {
+      expect(await workingBalanceHelper.getVotingEscrow()).to.equal(votingEscrow.address);
+    });
+
+    it('indicates where to read supply from', async () => {
+      expect(await workingBalanceHelper.readsTotalSupplyFromVE()).to.be.true;
+    });
+  });
+
+  before('setup accounts', async () => {
+    [, veBALHolder] = await getSigners();
+
+    veBALHolder = await impersonate(veBALHolder.address, VAULT_BOUNTY.add(fp(5))); // plus gas
+    lpTokenHolder = await impersonate(LP_TOKEN_HOLDER, fp(100));
+  });
+
+  before('setup contracts', async () => {
+    vault = await new Task('20210418-vault', TaskMode.READ_ONLY, getForkedNetwork(hre)).deployedInstance('Vault');
+
+    veDelegationProxy = await new Task(
+      '20220325-ve-delegation',
+      TaskMode.READ_ONLY,
+      getForkedNetwork(hre)
+    ).deployedInstance('VotingEscrowDelegationProxy');
+
+    votingEscrow = await new Task(
+      '20220325-gauge-controller',
+      TaskMode.READ_ONLY,
+      getForkedNetwork(hre)
+    ).deployedInstance('VotingEscrow');
+
+    const BALTokenAdmin = await new Task(
+      '20220325-balancer-token-admin',
+      TaskMode.READ_ONLY,
+      getForkedNetwork(hre)
+    ).deployedInstance('BalancerTokenAdmin');
+
+    BAL = await BALTokenAdmin.getBalancerToken();
+    gauge = await task.instanceAt('LiquidityGaugeV5', GAUGE);
+    lpToken = await deployedAt('IERC20', LP_TOKEN);
+  });
+
+  context('with no veBAL', () => {
+    const stakeAmount = fp(100);
+
+    before('stake in gauge', async () => {
+      await lpToken.connect(lpTokenHolder).transfer(veBALHolder.address, stakeAmount);
+      await lpToken.connect(veBALHolder).approve(gauge.address, MAX_UINT256);
+
+      await gauge.connect(veBALHolder)['deposit(uint256)'](stakeAmount);
+    });
+
+    it('projected balance should equal current', async () => {
+      const [currentWorkingBalance, projectedWorkingBalance] = await workingBalanceHelper.getWorkingBalances(
+        gauge.address,
+        veBALHolder.address
+      );
+
+      // Ensure we have equal balances (that are non-zero)
+      expect(projectedWorkingBalance).to.eq(currentWorkingBalance);
+      expect(projectedWorkingBalance).to.gt(0);
+    });
+  });
+
+  context('with veBAL', () => {
+    const TOKENLESS_PRODUCTION = 0.4;
+    const MAX_BALANCE_RATIO = 1 / TOKENLESS_PRODUCTION;
+
+    before('create veBAL whale', async () => {
+      bal80weth20Pool = await deployedAt('v2-pool-weighted/WeightedPool', VEBAL_POOL);
```

Comment:
You should use the corresponding task for this, not the artifact from the live package.

Reply:
Well, I took this from the Gnosis Root Gauge Factory deployment. I need that exact pool, not just any pool.

I just call `getPoolId` and `approve` on it though; I don't need it "as" a WeightedPool. So what I can do is hard-code the poolId instead of the address. Then I can just get it as an IERC20 to call approve.

const [poolAddress] = await vault.getPool(VEBAL_POOL_ID);
const bal80weth20Pool = await deployedAt('IERC20', poolAddress);